instrumentation/spring/spring-web/spring-web-3.1/library/src/main/java/io/opentelemetry/instrumentation/spring/web/SpringWebHttpAttributesGetter.java

```diff
@@ -41,13 +43,60 @@ public String flavor(HttpRequest httpRequest, @Nullable ClientHttpResponse clien
     return null;
   }
 
+  @Nullable private static final MethodHandle GET_STATUS_CODE;
+
+  @Nullable private static final MethodHandle STATUS_CODE_VALUE;
+
+  static {
+    MethodHandle getStatusCode = null;
+    MethodHandle statusCodeValue = null;
+    Class<?> httpStatusCodeClass = null;
+
+    MethodHandles.Lookup lookup = MethodHandles.publicLookup();
+
+    try {
+      httpStatusCodeClass = Class.forName("org.springframework.http.HttpStatusCode");
+    } catch (ClassNotFoundException e) {
+      try {
+        httpStatusCodeClass = Class.forName("org.springframework.http.HttpStatus");
+      } catch (ClassNotFoundException ignored) {
+        // ignored
+      }
+    }
+
+    if (httpStatusCodeClass != null) {
+      try {
+        getStatusCode =
+            lookup.findVirtual(
+                ClientHttpResponse.class,
+                "getStatusCode",
+                MethodType.methodType(httpStatusCodeClass));
+        statusCodeValue =
+            lookup.findVirtual(httpStatusCodeClass, "value", MethodType.methodType(int.class));
+      } catch (NoSuchMethodException | IllegalAccessException ignored) {
+        // ignored
+      }
+    }
+
+    GET_STATUS_CODE = getStatusCode;
+    STATUS_CODE_VALUE = statusCodeValue;
+  }
+
   @Override
   public Integer statusCode(
       HttpRequest httpRequest, ClientHttpResponse clientHttpResponse, @Nullable Throwable error) {
+
+    if (GET_STATUS_CODE == null || STATUS_CODE_VALUE == null) {
+      return null;
+    }
+
     try {
-      return clientHttpResponse.getStatusCode().value();
+      Object statusCode = GET_STATUS_CODE.invoke(clientHttpResponse);
+      return (int) STATUS_CODE_VALUE.invoke(statusCode);
     } catch (IOException e) {
-      return HttpStatus.INTERNAL_SERVER_ERROR.value();
+      return 500;
```

Comment:
I'd vote for removing the `IOException` catch, and just returning `null` if we can't retrieve the status code